packages/modular-scripts/src/esbuild-scripts/plugins/rewriteDependenciesPlugin.ts

```diff
@@ -0,0 +1,140 @@
+import * as esbuild from 'esbuild';
+import type { Dependency } from '@schemastore/package';
+
+export function createRewriteDependenciesPlugin(
+  externalDependencies: Dependency,
+  target?: string[],
+): esbuild.Plugin {
+  const externalCdnTemplate =
+    process.env.EXTERNAL_CDN_TEMPLATE ??
+    'https://cdn.skypack.dev/[name]@[version]';
+
+  const importMap: Record<string, string> = Object.entries(
+    externalDependencies,
+  ).reduce(
+    (acc, [name, version]) => ({
+      ...acc,
+      [name]: externalCdnTemplate
+        .replace('[name]', name)
+        .replace('[version]', version),
+    }),
+    {},
+  );
```

Comment:
nit - this has potential for an `Object.prototype` vector as the `importMap` is used with `in` below.

Consider switching to a `Map` or `Object.create(null)`:

```
  const importMap: Map<string, string> = new Map(Object.entries(
    externalDependencies,
  ).map(
    ([name, version]) => [name, externalCdnTemplate
        .replace('[name]', name)
        .replace('[version]', version)]
  ));

// later
if (importMap.has(dependencyName)) {
```